controllers/internal/status/components.go

```diff
@@ -0,0 +1,96 @@
+package status
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/kverrors"
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+
+	"github.com/ViaQ/loki-operator/internal/external/k8s"
+	"github.com/ViaQ/loki-operator/internal/manifests"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetComponentsStatus updates the pod status map component
+func SetComponentsStatus(ctx context.Context, k k8s.Client, req ctrl.Request) error {
+	var s lokiv1beta1.LokiStack
+	if err := k.Get(ctx, req.NamespacedName, &s); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return kverrors.Wrap(err, "failed to lookup lokistack", "name", req.NamespacedName)
+	}
+
+	var (
+		compactorName     = manifests.CompactorName(s.Name)
+		distributorName   = manifests.DistributorName(s.Name)
+		ingesterName      = manifests.IngesterName(s.Name)
+		querierName       = manifests.QuerierName(s.Name)
+		queryFrontendName = manifests.QueryFrontendName(s.Name)
+		psm               = lokiv1beta1.PodStatusMap{}
+	)
+	type componentMapping struct {
+		label string
+		psm   lokiv1beta1.PodStatusMap
+	}
+	components := map[string]componentMapping{
+		compactorName: {
+			label: manifests.LabelCompactorComponent,
+			psm:   psm.DeepCopy(),
```

Comment:
What's the point of using DeepCopy? I think you should just use a `lokiv1beta1.PodStatusMap{}` since `psm` is an empty struct. It looks like you're only using `psm` for this reason.